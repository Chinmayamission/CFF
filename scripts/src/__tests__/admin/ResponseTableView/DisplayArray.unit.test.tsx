import { render } from "enzyme";
import React from "react";
import ResponseTableView from "../../../admin/ResponseTable/ResponseTableView";
import { createResponseWithValue } from "../constants";

it("displays array value properly", () => {
  const dataOptionView = {
    id: "all",
    displayName: "All Responses",
    columns: ["volunteer"]
  };
  const renderedForm = {
    schema: {
      type: "object",
      properties: {
        volunteer: { type: "array", items: { type: "string" } }
      }
    },
    uiSchema: null,
    formOptions: null,
    name: "",
    _id: null
  };
  const responses = [
    createResponseWithValue({ volunteer: ["Teaching", "Dancing"] })
  ];
  const wrapper = render(
    <ResponseTableView
      responses={responses}
      renderedForm={renderedForm}
      dataOptionView={dataOptionView}
    />
  );
  expect(wrapper).toMatchSnapshot();
  // expect(wrapper.text()).toContain("Adults Display Name");
  // expect(wrapper.find("select").text()).toContain("Teaching");
  // expect(wrapper.find("select").text()).toContain("Dancing");
  expect(wrapper.text()).not.toContain('["Teaching","Dancing"]');
  expect(wrapper.text()).toContain("Teaching, Dancing");
});
