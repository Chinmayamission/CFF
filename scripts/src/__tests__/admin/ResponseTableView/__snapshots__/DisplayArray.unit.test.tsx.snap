// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays array value properly 1`] = `
<div>
  <button
    class="btn btn-outline-primary"
  >
    Download CSV
  </button>
  <div
    class="ReactTable"
  >
    <div
      class="rt-table"
      role="grid"
    >
      <div
        class="rt-thead -header"
        style="min-width:100px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Volunteer
            </div>
            <div
              class="rt-resizer"
            />
          </div>
        </div>
      </div>
      <div
        class="rt-thead -filters"
        style="min-width:100px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="rt-tbody"
        style="min-width:100px"
      >
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              Teaching, Dancing
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination-bottom"
    >
      <div
        class="-pagination"
      >
        <div
          class="-previous"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          class="-center"
        >
          <span
            class="-pageInfo"
          >
            Page 
            <div
              class="-pageJump"
            >
              <input
                aria-label="jump to page"
                type="number"
                value="1"
              />
            </div>
             of 
            <span
              class="-totalPages"
            >
              1
            </span>
          </span>
          <span
            class="select-wrap -pageSizeOptions"
          >
            <select
              aria-label="rows per page"
            >
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                selected=""
                value="20"
              >
                20 rows
              </option>
              <option
                value="25"
              >
                25 rows
              </option>
              <option
                value="50"
              >
                50 rows
              </option>
              <option
                value="100"
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          class="-next"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="-loading"
    >
      <div
        class="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
